Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -434,27 +433,37 @@ public WeakPropertyChangedProxy(INotifyPropertyChanged source, PropertyChangedEv
 			public void SubscribeTo(INotifyPropertyChanged source, PropertyChangedEventHandler listener)
 			{ 
 				source.PropertyChanged += _handler;
+				var bo = source as BindableObject;
+				if (bo != null)
+					bo.BindingContextChanged += _bchandler;
 				_source.SetTarget(source);
 				_listener.SetTarget(listener);
 			}
 
 			public void Unsubscribe()
 			{
-				INotifyPropertyChanged source;
-				if (_source.TryGetTarget(out source) && source!=null)
+				if (_source.TryGetTarget(out INotifyPropertyChanged source) && source != null)
 					source.PropertyChanged -= _handler;
+				var bo = source as BindableObject;
+				if (bo != null)
+					bo.BindingContextChanged -= _bchandler;
+
 				_source.SetTarget(null);
 				_listener.SetTarget(null);
 			}
 
 			void OnPropertyChanged(object sender, PropertyChangedEventArgs e)
 			{
-				PropertyChangedEventHandler handler;
-				if (_listener.TryGetTarget(out handler) && handler != null)
+				if (_listener.TryGetTarget(out PropertyChangedEventHandler handler) && handler != null)
 					handler(sender, e);
 				else
 					Unsubscribe();
 			}
+
+			void OnBCChanged(object sender, EventArgs e)
+			{
+				//OnPropertyChanged(sender, new PropertyChangedEventArgs("BindingContext"));
```

Comment:
Commented code?

Reply:
oups. that's wrong. the code should be in